scripts/west_commands/build.py

```diff
@@ -458,6 +458,11 @@ def _run_cmake(self, board, origin, cmake_opts):
             cmake_opts = ['-DBOARD={}'.format(board)]
         else:
             cmake_opts = []
+
+        gdb = config_get('gdb', None)
```

Comment:
Shouldn't this be done as part of the common runner code? where the existing `--gdb` option is parsed

Reply:
https://github.com/zephyrproject-rtos/zephyr/blob/main/scripts/west_commands/runners/core.py#L276

Reply:
I don't think so, why?

CMAKE_GDB is what gets used by the build system to set the `config: gdb:` value in build/zephyr/runners.yaml. This value is set at cmake time and provides the default gdb used by all the runners.

You would want to be able to `west config build.gdb path/to/arch1/gdb-py`, then do a bunch of builds, then `west config build.gdb path/to/arch2/gdb-py`, then do a bunch more builds for a different architecture, and have the default gdbs in each build directory refer to the correct GDB for the target architecture in each case.

Am I missing something?